src/main/java/org/opensearch/knn/index/codec/KNN9120Codec/DerivedSourceStoredFieldsWriter.java

```diff
@@ -0,0 +1,117 @@
+import java.util.Map;
+import java.util.Objects;
+
+@RequiredArgsConstructor
+public class DerivedSourceStoredFieldsWriter extends StoredFieldsWriter {
+
+    private final StoredFieldsWriter delegate;
+    private final List<String> vectorFieldTypes;
+
+    @Override
+    public void startDocument() throws IOException {
+        delegate.startDocument();
+    }
+
+    @Override
+    public void writeField(FieldInfo fieldInfo, int i) throws IOException {
+        delegate.writeField(fieldInfo, i);
+    }
+
+    @Override
+    public void writeField(FieldInfo fieldInfo, long l) throws IOException {
+        delegate.writeField(fieldInfo, l);
+    }
+
+    @Override
+    public void writeField(FieldInfo fieldInfo, float v) throws IOException {
+        delegate.writeField(fieldInfo, v);
+    }
+
+    @Override
+    public void writeField(FieldInfo fieldInfo, double v) throws IOException {
+        delegate.writeField(fieldInfo, v);
+    }
+
+    @Override
+    public void writeField(FieldInfo info, DataInput value, int length) throws IOException {
+        delegate.writeField(info, value, length);
+    }
+
+    @Override
+    public int merge(MergeState mergeState) throws IOException {
+        return delegate.merge(mergeState);
+    }
+
+    @Override
+    public void writeField(FieldInfo fieldInfo, BytesRef bytesRef) throws IOException {
+        // Parse out the vectors from the source
+        if (Objects.equals(fieldInfo.name, SourceFieldMapper.NAME) && !vectorFieldTypes.isEmpty()) {
+            Tuple<? extends MediaType, Map<String, Object>> mapTuple = XContentHelper.convertToMap(
+                BytesReference.fromByteBuffer(ByteBuffer.wrap(bytesRef.bytes)),
+                true,
+                MediaTypeRegistry.JSON
+            );
+            Map<String, Object> filteredSource = XContentMapValues.filter(null, vectorFieldTypes.toArray(new String[0]))
+                .apply(mapTuple.v2());
+            BytesStreamOutput bStream = new BytesStreamOutput();
+            MediaType actualContentType = mapTuple.v1();
+            XContentBuilder builder = MediaTypeRegistry.contentBuilder(actualContentType, bStream).map(filteredSource);
+            builder.close();
+            BytesReference bytesReference = bStream.bytes();
+            delegate.writeField(fieldInfo, bytesReference.toBytesRef());
+            return;
+        }
+        delegate.writeField(fieldInfo, bytesRef);
+    }
+
+    @Override
+    public void writeField(FieldInfo fieldInfo, String s) throws IOException {
+        delegate.writeField(fieldInfo, s);
+    }
+
+    @Override
+    public void finishDocument() throws IOException {
+        delegate.finishDocument();
+    }
+
+    @Override
+    public void finish(int i) throws IOException {
+        delegate.finish(i);
+    }
+
+    @Override
+    public void close() throws IOException {
+        delegate.close();
+    }
+
+    @Override
+    public long ramBytesUsed() {
+        return delegate.ramBytesUsed();
```

Comment:
Please add the bytes used by the member variables of this class in ramBytes

Reply:
Sure - let me see try to do that.

Reply:
Will add in follow up. Only other member is list of strings which are fields which will be on order of bytes to low kbs